src/index.js

```diff
@@ -1,81 +1,99 @@
+import PropTypes from 'prop-types';
 
-const Collapsable = ({
+const Collapsible = ({
+  isOpen,
+  minAnimationDuration,
+  maxAnimationDuration,
+  speedDivider,
+  easing,
+  children,
+  keepChildrenOnClose,
+  ...restProps
+}) => {
+  const content = useRef();
+  const [state, setState] = useState({
+    height: isOpen ? 'auto' : 0,
+    speed: 0,
+  });
+
+  const [effectiveChildren, setChildren] = useState(children);
+
+  const handleClose = () => {
+    if (!isOpen) {
+      content.current.style.visibility = 'hidden';
+
+      if (!keepChildrenOnClose) {
+        // remove child
+        setChildren(null);
+      }
+    }
+  };
+
+  useEffect(() => {
+    if (isOpen) {
+      const contentHeight = content.current.scrollHeight;
+      const time = contentHeight / speedDivider;
+      const animation =
+        time < minAnimationDuration
+          ? minAnimationDuration
+          : time > maxAnimationDuration
+          ? maxAnimationDuration
+          : time;
+      content.current.style.visibility = 'visible';
+      setChildren(children);
+      setState({
+        ...state,
+        height: contentHeight,
+        speed: animation,
+      });
+    } else {
+      setState({
+        ...state,
+        height: 0,
+      });
+    }
+  }, [
     isOpen,
     minAnimationDuration,
     maxAnimationDuration,
     speedDivider,
-    easing,
-    children
-}) => {
-    const content = useRef()
-    const [state, setState] = useState({
-        height: isOpen ? 'auto' : 0,
-        speed: 0
-    })
-
-    useEffect(() => {
-        if (isOpen) {
-            const contentHeight = content.current.scrollHeight
-            const time = contentHeight / speedDivider
-            const animation =
-                time < minAnimationDuration
-                    ? minAnimationDuration
-                    : time > maxAnimationDuration
-                    ? maxAnimationDuration
-                    : time
-            content.current.style.visibility = 'visible'
-            setState({
-                ...state,
-                height: contentHeight,
-                speed: animation
-            })
-        } else {
-            setState({
-                ...state,
-                height: 0
-            })
-            setTimeout(() => {
-                content.current.style.visibility = 'hidden'
-            }, state.speed * speedDivider)
-        }
-    }, [
-        isOpen,
-        minAnimationDuration,
-        maxAnimationDuration,
-        speedDivider,
-        children
-    ])
+    children,
+    effectiveChildren,
+  ]);
 
-    return (
-        <div
-            style={{
-                overflow: 'hidden',
-                height: state.height,
-                transition: `height ${state.speed}s ${easing}`
-            }}>
-            <div ref={content} style={{ overflow: 'auto' }}>
-                {children}
-            </div>
-        </div>
-    )
-}
+  return (
+    <div
+      {...restProps}
```

Comment:
This could lead to invalid props being passed down. I'm not sure it's needed as for custom styling you can just wrap it in a custom component which has it's own styling etc..

Reply:
Yeah I feel allowing for users to add their own class name to the wrapper would suffice.